pilot/pkg/model/authentication.go

```diff
@@ -97,26 +92,29 @@ func (policy *AuthenticationPolicies) addRequestAuthentication(configs []Config)
 func (policy *AuthenticationPolicies) GetJwtPoliciesForWorkload(namespace string,
 	workloadLabels labels.Collection) []*Config {
 	configs := make([]*Config, 0)
-	if nsConfig, ok := policy.requestAuthentications[namespace]; ok {
-		for idx := range nsConfig {
-			cfg := &nsConfig[idx]
-			if namespace != cfg.Namespace {
-				// Should never come here. Log warning just in case.
-				log.Warnf("Seeing config %s with namespace %s in map entry for %s. Ignored", cfg.Name, cfg.Namespace, namespace)
-				continue
-			}
-			spec := cfg.Spec.(*v1beta1.RequestAuthentication)
-			selector := labels.Instance(spec.GetSelector().GetMatchLabels())
-			if workloadLabels.IsSupersetOf(selector) {
-				configs = append(configs, cfg)
+	lookupInNamespaces := []string{namespace}
+	if namespace != policy.rootNamespace {
+		// Only check the root namespace if the (workload) namespace is not already the root namespace
+		// to avoid double inclusion.
+		lookupInNamespaces = append(lookupInNamespaces, policy.rootNamespace)
```

Comment:
> validation webhook does not have root config namespace information,

what does this policy.rootnamespace mean then?

Reply:
This is at run time (request), on pilot.

What I mean is the validation cannot reject policy with select in root namespace as I wanted. For alpha policy, global policy is in cluster scope, so it is distinguished from the rest in the validation flow. Changing to the root-namespace model, we lost that capability, so we need to check all policies in the root namespace as usual. This is what other APIs behave today anyway.